packages/flutter/lib/src/material_animated_icons/animated_icons.dart

```diff
@@ -0,0 +1,269 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+part of material_animated_icons;
+
+// The code for drawing animated icons is kept in a private API, as we are not
+// yet ready for exposing a public API for (partial) vector graphics support.
+// See: https://github.com/flutter/flutter/issues/1831 for details regarding
+// generic vector graphics support in Flutter.
+
+/// Shows an animated icon at a given animation [progress].
+///
+/// The available icons are specified in [AnimatedIcons].
+class AnimatedIcon extends StatelessWidget {
+
+  /// Creates an AnimatedIcon.
+  ///
+  /// [progress], and [icon] cannot be null.
+  /// The [size] and [color] default to the value given by the current [IconTheme].
+  const AnimatedIcon({
+    Key key,
+    @required this.progress,
+    @required this.icon,
+    this.color,
+    this.size,
+    this.semanticLabel,
+    this.textDirection,
+    // TODO(amirh): add a parameter for controlling scaling behavior.
+  }) : assert(progress != null),
+       assert(icon != null);
+
+  /// The animation progress for the animated icon.
+  /// The value is clamped to be between 0 and 1.
```

Comment:
needs a blank line otherwise the second line will be considered part of the first line's paragraph

Reply:
done